tests/core/providers/test_auto_provider.py

```diff
@@ -145,11 +145,12 @@ def test_web3_auto_infura_with_secret(monkeypatch, caplog, environ_name):
     monkeypatch.setenv(environ_name, 'test')
     monkeypatch.setenv('WEB3_INFURA_API_SECRET', 'secret')
 
-    importlib.reload(infura)
+    with pytest.raises(DeprecationWarning):
```

Comment:
Per previous discussion, I'm no longer deprecating the `kwargs` because they are still needed for params passed to each `post` (`timeout`, specifically).